packages/frontend/src/components/UnderlyingData/UnderlyingDataModalContent.tsx

```diff
@@ -147,13 +150,23 @@ const UnderlyingDataModalContent: FC<Props> = () => {
             },
         };
 
+        const showUnderlyingValues: string[] | undefined =
+            isField(meta?.item) && isMetric(meta.item)
+                ? meta.item.showUnderlyingValues
+                : undefined;
+
         const availableDimensions = allDimensions.filter(
             (dimension) =>
                 tablesInQuery.has(dimension.table) &&
                 !dimension.timeInterval &&
-                !dimension.hidden,
+                !dimension.hidden &&
+                (showUnderlyingValues !== undefined
+                    ? showUnderlyingValues.includes(dimension.name)
```

Comment:
I can see a problem where multiple tables have the same dimension name. 

So in my list I have `id` it would include all other them instead of the one from the same table as the metric.